pkg/gameserverallocations/controller.go

```diff
@@ -65,25 +71,100 @@ type Controller struct {
 	gameServerAllocationSynced cache.InformerSynced
 	gameServerAllocationGetter getterv1alpha1.GameServerAllocationsGetter
 	stop                       <-chan struct{}
-	allocationMutex            *sync.Mutex
 	workerqueue                *workerqueue.WorkerQueue
+	gsWorkerqueue              *workerqueue.WorkerQueue
 	recorder                   record.EventRecorder
+	readyGameServers           gameServerCacheEntry
+	topNGameServerCnt          int
+}
+
+// gameserver cache to keep the Ready state gameserver.
+type gameServerCacheEntry struct {
+	mu    sync.RWMutex
+	cache map[string]*v1alpha1.GameServer
+}
+
+// LoadOrStore returns data if exists or stores if it is not. Return the value and true if the data already exists.
+func (e *gameServerCacheEntry) LoadOrStore(key string, gs *v1alpha1.GameServer) (*v1alpha1.GameServer, bool) {
```

Comment:
Is this here to make testing easier, I assume?

Reply:
I originally used [syncmap](https://godoc.org/golang.org/x/sync/syncmap#Map.Range). Later on I decided to use my own synced cache so instead of changing my code, I implemented the syncmap interface. This was I can go back using syncmap as well.

Reply:
Fair enough - although removing methods is not going to block you from using sync.Map :)

I just saw you used LoadOrStore in your tests, so figured it was easier for testing :+1: